README.md

```diff
@@ -1,81 +1,119 @@
 
-# LSP overview
+## LSP overview
 
-Neovim supports the Language Server Protocol (LSP), which means it acts as a client to language servers and includes a Lua framework `vim.lsp` for building enhanced LSP tools. LSP facilitates features like:
+Neovim supports the [Language Server Protocol (LSP)](https://microsoft.github.io/language-server-protocol/), which means it acts as a client to language servers and includes a Lua framework (`vim.lsp`) for building enhanced LSP tools.
+LSP facilitates features like:
 
 - go-to-definition
 - find-references
 - hover
 - completion
 - rename
-- format 
+- format
 - refactor
 
-Neovim provides an interface for all of these features, and the language server client is designed to be highly extensible to allow plugins to integrate language server features which are not yet present in Neovim core such as [**auto**-completion](https://github.com/neovim/nvim-lspconfig/wiki/Autocompletion) (as opposed to manual completion with omnifunc) and [snippet integration](https://github.com/neovim/nvim-lspconfig/wiki/Snippets).
+Neovim itself provides an interface for all of these features.
+The LSP client is designed to be highly extensible; it allows plugins to integrate language server features which are not yet present in Neovim core.
+Examples of language server features not yet in Neovim core:
 
-These features are not implemented in this repo, but in Neovim core. See `:help lsp` for more details.
+- [**auto**-completion](https://github.com/neovim/nvim-lspconfig/wiki/Autocompletion) (as opposed to manual completion using [omnifunc](https://neovim.io/doc/user/options.html#'omnifunc'))
+- [snippet integration](https://github.com/neovim/nvim-lspconfig/wiki/Snippets).
 
 ## Install
 
-* Requires [Neovim HEAD/nightly](https://github.com/neovim/neovim/releases/tag/nightly) (v0.5 prerelease). Update Neovim and nvim-lspconfig before reporting an issue.
+- Requires [Neovim HEAD/nightly](https://github.com/neovim/neovim/releases/tag/nightly) (v0.5 prerelease). Update Neovim and nvim-lspconfig before reporting an issue.
+- Install nvim-lspconfig like any other plugin, e.g., using [vim-plug](https://github.com/junegunn/vim-plug):
 
-* Install nvim-lspconfig like any other Vim plugin, e.g. with [vim-plug](https://github.com/junegunn/vim-plug):
 ```vim
 :Plug 'neovim/nvim-lspconfig'
 ```
 
 ## Quickstart
-1. Install a language server, e.g. [pyright](CONFIG.md#pyright)
+
+1. Install a language server, e.g., [pyright](CONFIG.md#pyright)
+
 ```bash
 npm i -g pyright
 ```
-2. Add the language server setup to your init.vim. The server name must match those found in the table of contents in [CONFIG.md](CONFIG.md)
+
+1. Add the language server setup to your `init.vim`.
+   The server name must match those found in the table of contents in [CONFIG.md](CONFIG.md).
+   _NOTE_: this snippet is for `init.vim` only, this will not work in a `init.lua` file.
```

Comment:
It will, if you remove the heredoc, we should make that clear without getting too wordy